### ISSUE TITLE
checkboxlist within a RadioMultiAnswer

### ISSUE THREAD

**Issue by Alex-Jordan** [openwebwork/pg#1007]
I'm pushing the limits of `parserRadioMultiAnswer.pl` with the problem below. It renders fine, but there are two issues:

- When you have not selected the second radio button (which contains checkboxes), only the first checkbox is grayed out. I think the intent is that all would be grayed out until you select that radio button.
- When I submit the correct answer (which is indeed the correct answer as displayed by show correct answers) it is not accepted as correct.

Does it go beyond the scope of `parserRadioMultiAnswer.pl` to use checkboxes within an option?

```
# WeBWorK problem written by Alex Jordan 
DOCUMENT();

loadMacros(qw(
    PGstandard.pl PGML.pl PGtikz.pl parserCheckboxList.pl parserRadioMultiAnswer.pl
));

$x = list_random(-2,2);
$y = list_random(-2,2);

$image = createTikZImage();
$image->BEGIN_TIKZ
\fill[color=blue!20] (0,0) -| ($x,$y) -| cycle;
\draw[{stealth}-{stealth}] (-2,0) -- (2,0);
\draw[{stealth}-{stealth}] (0,-2) -- (0,2);
\draw[-{stealth}, ultra thick, color=blue] (0,0) -- ($x,0);
\draw[-{stealth}, ultra thick, color=blue] (0,0) -- (0,$y);
\node at ($x/2,$y/2) {\(H\)};
END_TIKZ

$checks = CheckboxList(
    [
        '\(\vec{0}\) is not in \(H\)',
        'There are vectors \(\vec{u}\) and \(\vec{v}\) in \(H\) with \(\vec{u}+\vec{v}\) not in \(H\).',
        'There is a vector \(\vec{u}\) in \(H\) and a scalar \(c\) in \(\mathbb{R}\) with \(c\vec{u}\) not in \(H\).',
    ],
    [2]
);

$rma = RadioMultiAnswer(
    [
        ['\(H\) is a subspace.'],
        ['\(H\) is not a subspace because (check all that apply): %s', $checks]
    ],
    1
)->with(labels => ['Yes','No']);

BEGIN_PGML
In the image below, a subset [`H`] of [`\mathbb{R}^2`] is shaded. Is [`H`] a subspace of [`\mathbb{R}^2`]?

>> [! R^2 with one quadrant and its boundary shaded !]{$image} <<

[_]{$rma}
END_PGML

ENDDOCUMENT();
```

**Comment by Alex-Jordan** [comment]
What I described is in 2.18.

I just checked devleop, and the graying out issue is not an issue there. However it still does not accept the correct answer selections.

**Comment by drgrice1** [comment]
The issue with the checkboxes not graying out correctly was fixed in https://github.com/openwebwork/pg/pull/930.

It turns out I also knew about the other issue, but I forgot about it.  I also don't know how to fix it so that the following workaround is not needed.  What you need to do is add `checkTypes => 0` to the `RadioMultiAnswer` object.  So that would be
```perl
$rma = RadioMultiAnswer(
	[
		['\(H\) is a subspace.'],
		[ '\(H\) is not a subspace because (check all that apply): %s', $checks ] ],
	1,
	checkTypes => 0,
	labels     => [ 'Yes', 'No' ]
);
```
Note that this is only needed when a `CheckboxList` is used as a part answer, and the `CheckboxList` only has one correct answer.  The `checkTypes => 0` option is not needed if there are more than one correct answer.  (Also note that you don't need to use `with` to add options to a `RadioMultiAnswer`.)

Also note that the same thing is true for a `CheckboxList` part answer of a `MutliAnswer` object from the `parserMultiAnswer.pl` macro.  There you have to provide a custom checker, but the checker doesn't even get a chance to check the answers unless you set `checkTypes => 0` when a `CheckboxList` has only one correct answer.

The issue is that the correct answer always has type "List".  A student answer is of type "String" in the case that only one checkbox is checked, but is also of type "List" if multiple checkboxes are checked.

**Comment by pstaabp** [comment]
Fixed in #1011
